### ISSUE TITLE
What versions of Ubuntu is this compatible with? Big problems with bootable drive attempt

### ISSUE THREAD

**Issue by thedonquixotic** [jcjohnson/neural-style#258]
Hey, 

I'm an artist/designer. I know a little code, but I am no terminal savvy code wiz by far. I've been experimenting with code based tools though and expanding my tool set and thought I'd dip into some neural style stuff. 

Running into some problems. 

I first tried running this thing on Windows (well actually tried a python port that seemed easier for windows) but ran into so many problems I finally decided to go with a bootable Linux solution. 

Created a bootable persistent linux usb. Got torch and all it's dependcies and neural style installed: you can see more [[here]](https://groups.google.com/forum/#!msg/torch7/1Q38zSpWf4o/R17q_qo3AgAJ) 

But 4 gigs wasn't enough once I wanted to install CUDA so I had to start over and try and make a bootable drive with a larger casper-rw using this method found [[here]](http://www.pendrivelinux.com/create-a-larger-than-4gb-casper-partition/)

That didn't work. I went through a bunch of different things. You can see one of my midway attempts at fixing the problem [[here]](http://askubuntu.com/questions/789404/persistent-linux-usb-wont-boot-getting-weird-error-screens-maybe-related-to?noredirect=1#comment1186694_789404)

One of the tutorials I saw specifically mentioned that they were using 13.1 I saw that version number repeated a couple of times, so I'm thinking maybe I need to rever from 14.1 to 13.1? Delving into a bunch of help threads and stuff it seems there's some wonky stuff with persistent booting in 14 onward (I believe it was totally removed in 15?)

I really really want to get this working. I've been hacking away at this solid for 3 days. If I can get this working I want to put up an artist friendly tutorial on Medium for other non-coder types to use. I really just need a little more of a nudge to get over this hill. 

**TL;DR**
- Can I use this neural network on Ubuntu 13.1 persistent boot drive? Is it compatible?
- What are other basic code dependcies not listed? Over the course of installing this I found I needed to install build-essentials, gcc, curl, and a couple of other things that were probably not included because they're pretty much standard for any coder. I want to be systemic on my note taking for this so I can list this stuff in my article.

**Comment by Runescaped** [comment]
I recommend using 14.04, because CUDA 7.5 was officially built for Ubuntu 14.04 and 15.04, and CUDA 8.0 was built for Ubuntu 14.04 and 16.04.

That being said, I've never tried to get this to work w/ 13.10...

**Comment by jcjohnson** [comment]
I use Ubuntu 14.04, so I guess that's the one that is "officially"
supported. However everything should work as long as you can get CUDA
installed.

**Comment by thedonquixotic** [comment]
Thank you for the help. 

Cuda 6.5 works with 13 I believe. I've tried doing 14.04, but I just can't seem to get a persistent install of it working on my usb drive. I don't know why. I think I might just try setting up a Linux dual boot on one of my internal hdd. I wanted to avoid that in order to make this more portable, and maybe be easier to spread, but I feel like it might be more trouble than it's worth.

**Comment by Runescaped** [comment]
> I've tried doing 14.04, but I just can't seem to get a persistent install of it working on my usb drive. I don't know why.

I was able to get a persistent 14.04 install working on my usb before, but I didn't finish the CUDA installation. Right now I'm trying to get everything running on an 8gb usb drive, I'll let you know how that turns out.

Currently I run Ubuntu off a partition on a hard drive that I had lying around. I also didn't feel like dealing w/ a dual booting configuration.

**Comment by thedonquixotic** [comment]
Yeah I got it working on the 14 working on a persistent drive too, but only without CUDA, since CUDA is too big for the 4 gig limit. Good luck on getting the 8 gig drive to work! Would be pretty fun to have a neural network working on a USB haha!

**Comment by thedonquixotic** [comment]
hey @Runescaped so I created a dual boot partition and got this working. I thought I'd let you know that after installing CUDA, the OS and NN took up more than 10 gigs total so even if you get the persistent USB boot working, I think you will probably need something bigger than 8 gigs.

**Comment by Runescaped** [comment]
> the OS and NN took up more than 10 gigs total so even if you get the persistent USB boot working, I think you will probably need something bigger than 8 gigs.

Yeah, I sort of suspected I'd still have some issues installing everything. But the main problem I encountered was the same  one you had. I was able to get a persistent install, but when I created a partition named casper-rw and deleted the casper-rw file, Ubuntu either stopped being persistent, or booted up to that BusyBox prompt that you experienced. Currently it appears that getting everything setup via usb just isn't a viable option.

**Comment by thedonquixotic** [comment]
done
